### PR TITLE
Refactoring ConstraintValidators to sanitize violation message templates

### PR DESCRIPTION
**Pull request by joshi-keyur**
### Description of the Change

Sanitize message string so it is not an EL expression by escaping special characters.
The message string in this case is a collection of user supplied values which is being used as a part of message template for message interpolation.

### CODE REVIEW

File: titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/sanitizer/SchedulingConstraintSetValidator.java [hunk 1]
```diff
@@ -53,7 +54,7 @@ public void initialize(SchedulingConstraintSet constraintAnnotation) {
     }
 
     @Override
-    public boolean isValid(Container container, ConstraintValidatorContext context) {
+    public boolean isValid(Container container, ConstraintValidatorContextWrapper context) {
```
Comment (tbak) [inline titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/sanitizer/SchedulingConstraintSetValidator.java]: `protected`? We want clients to call the original validator method?

File: titus-common/src/main/java/com/netflix/titus/common/model/sanitizer/internal/AbstractConstraintValidator.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import java.lang.annotation.Annotation;
+
+public abstract class AbstractConstraintValidator<A extends Annotation, T> implements ConstraintValidator<A, T> {
+
+    @Override
+    final public boolean isValid(T type, ConstraintValidatorContext context) {
+        return this.isValid(type, new ConstraintValidatorContextWrapper(context));
+    }
+
+    abstract public boolean isValid(T type, ConstraintValidatorContextWrapper wrapper);
```
Comment (tbak) [inline titus-common/src/main/java/com/netflix/titus/common/model/sanitizer/internal/AbstractConstraintValidator.java]: `protected`?

File: titus-common/src/main/java/com/netflix/titus/common/model/sanitizer/internal/ConstraintValidatorContextWrapper.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.common.model.sanitizer.internal;
+
+import javax.validation.ConstraintValidatorContext;
+
+public class ConstraintValidatorContextWrapper {
```
Comment (tbak) [inline titus-common/src/main/java/com/netflix/titus/common/model/sanitizer/internal/ConstraintValidatorContextWrapper.java]: Replace this class with `Function< String, ConstraintViolationBuilder >`, and the default implementation as here?

File: titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/sanitizer/SchedulingConstraintValidator.java [hunk 4]
```diff
@@ -66,17 +65,16 @@
 
     @Override
     public void initialize(SchedulingConstraint constraintAnnotation) {
-    }
+   }
```
Comment (fabiokung) [inline titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/sanitizer/SchedulingConstraintValidator.java]: minor: weird indentation here

File: titus-common/src/main/java/com/netflix/titus/common/model/sanitizer/internal/ConstraintValidatorContextWrapper.java (hunk 3 above)
Comment (joshi-keyur) [reply titus-common/src/main/java/com/netflix/titus/common/model/sanitizer/internal/ConstraintValidatorContextWrapper.java]: Thank you for the feedback. Please review the updates.

**Pull request merged**